integration/token/fungible/views/multisig.go

```diff
@@ -0,0 +1,235 @@
+	var tx *ttx.Transaction
+	txOpts := []ttx.TxOption{
+		ttx.WithTMSIDPointer(lv.TMSID),
+		ttx.WithAuditor(view2.GetIdentityProvider(context).Identity(lv.Auditor)),
+		ttx.WithAnonymousTransaction(!lv.NotAnonymous),
+	}
+	tx, err = ttx.NewTransaction(context, nil, txOpts...)
+	assert.NoError(err, "failed creating transaction")
+
+	// lock
+	err = multisig.Wrap(tx).Lock(senderWallet, lv.Type, lv.Amount, recipient)
+	assert.NoError(err, "failed adding transfer action [%d:%v]", lv.Amount, recipient)
+
+	_, err = context.RunView(ttx.NewCollectEndorsementsView(tx))
+	assert.NoError(err, "failed to sign transaction [<<<%s>>>]", tx.ID())
+
+	// Sanity checks:
+	// - the transaction is in pending state
+	owner := ttx.NewOwner(context, tx.TokenService())
+	vc, _, err := owner.GetStatus(tx.ID())
+	assert.NoError(err, "failed to retrieve status for transaction [%s]", tx.ID())
+	assert.Equal(ttx.Pending, vc, "transaction [%s] should be in busy state", tx.ID())
+
+	// Send to the ordering service and wait for finality
+	_, err = context.RunView(ttx.NewOrderingAndFinalityView(tx))
+	assert.NoError(err, "failed asking ordering")
+
+	// Sanity checks:
+	// - the transaction is in confirmed state
+	vc, _, err = owner.GetStatus(tx.ID())
+	assert.NoError(err, "failed to retrieve status for transaction [%s]", tx.ID())
+	assert.Equal(ttx.Confirmed, vc, "transaction [%s] should be in valid state", tx.ID())
+
+	return tx.ID(), nil
+}
+
+type MultiSigLockViewFactory struct{}
+
+func (f *MultiSigLockViewFactory) NewView(in []byte) (view.View, error) {
+	v := &MultiSigLockView{MultiSigLock: &MultiSigLock{}}
+	err := json.Unmarshal(in, v.MultiSigLock)
+	assert.NoError(err, "failed unmarshalling input")
+	return v, nil
+}
+
+type MultiSigRequestSpend struct{}
+
+func (m *MultiSigRequestSpend) Call(context view.Context) (interface{}, error) {
+	return nil, nil
+}
+
+// MultiSigSpend contains the input information to spend a token
+type MultiSigSpend struct {
+	// Auditor is the name of the auditor that must be contacted to approve the operation
+	Auditor string
+	// Wallet is the identifier of the wallet that owns the tokens to transfer
+	Wallet string
+	// Escrow contains the identities of the FSC nodes of the multisig escrow
+	Recipient view.Identity
+	// The TMS to pick in case of multiple TMSIDs
+	TMSID     *token2.TMSID
+	TokenType token.Type
+}
+
+type MultiSigSpendView struct {
+	*MultiSigSpend
+}
+
+func (r *MultiSigSpendView) Call(context view.Context) (res interface{}, err error) {
+	serviceOpts := ServiceOpts(r.TMSID)
+	recipient, err := ttx.RequestRecipientIdentity(context, r.Recipient, serviceOpts...)
+	assert.NoError(err, "failed getting recipient")
+
+	// choose the multisig token to spend
+	spendWallet := multisig.GetWallet(context, r.Wallet, serviceOpts...)
+	assert.NotNil(spendWallet, "wallet [%s] not found", r.Wallet)
+
+	matched, err := multisig.Wallet(context, spendWallet).ListTokens()
+	assert.NoError(err, "multisig script has expired")
```

Comment:
I don't think expiration is relevant here.

Reply:
okay